### DISCUSSION TITLE
[Deploy] How to deploy juicefs in k8s

### DISCUSSION THREAD

**Discussion by edenbuaa** [juicedata/juicefs#5061] [Q&A]
The deployment docs made me confused about how to deploy fuicesfs in my k8s environment.

Is there a full deployment yaml example to introduce how to deploy fuicesfs on k8s (assume that I have installed `redis `and `minio`)?

And how to use the fuse client pod to mount fuicesfs to `hostpath`?

**Comment by zhijian-pro** [comment]
https://juicefs.com/docs/community/how_to_use_on_kubernetes/

**Comment by edenbuaa** [comment]
@zhijian-pro thanks. I can deploy juicefs by the csi way， but I can't find where to config the cache method or directory.  There's a /var/jfsCache/ directory in the host, How to change it?

**Reply by zxh326** [reply]
https://juicefs.com/docs/csi/guide/cache

**Comment by a823883735** [comment]
How to integrate Prometheus when deploying in a Kubernetes cluster using the following method：
`# Kubernetes version >= v1.18
kubectl apply -f https://raw.githubusercontent.com/juicedata/juicefs-csi-driver/master/deploy/k8s.yaml`

**Reply by a823883735** [reply]
> 在Kubernetes集群中部署时集成Prometheus的方法如下： `# Kubernetes version >= v1.18 kubectl apply -f https://raw.githubusercontent.com/juicedata/juicefs-csi-driver/master/deploy/k8s.yaml`

metric 服务如何开启？

**Reply by zxh326** [reply]
> > 在Kubernetes集群中部署时集成Prometheus的方法如下： `# Kubernetes version >= v1.18 kubectl apply -f https://raw.githubusercontent.com/juicedata/juicefs-csi-driver/master/deploy/k8s.yaml`
> 
> metric 服务如何开启？

https://juicefs.com/docs/zh/csi/administration/going-production/#monitoring

默认设置下（未使用 hostNetwork），Mount Pod 通过 9567 端口提供监控 API（也可以通过在 mountOptions 中添加 metrics 选项来自定义端口号），端口名为 metrics。
